src/parse.rs

```diff
@@ -1096,7 +1096,7 @@ impl<'t, 'l, 'py> Parser<'t, 'l, 'py> {
             return Err(ParseError::EmptyVariable { at: at.into() });
         };
         let mut var = match variable_token.token_type {
-            VariableTokenType::Variable => self.parse_for_variable(variable_token.at).into(),
+            VariableTokenType::Variable(_) => self.parse_for_variable(variable_token.at).into(),
```

Comment:
We can get `at` from the `Variable` now:

```suggestion
            VariableTokenType::Variable(variable) => self.parse_for_variable(variable.at).into(),
```